Assets/_BForBoss/_Weapons/Scripts/EquipmentBehaviour.cs

```diff
@@ -0,0 +1,160 @@
+            {
+                _swapWeaponInputAction.Enable();
+            }
+
+            _isMouseScrollEnabled = true;
+        }
+        
+        private void DisableEquipmentPlayerInput()
+        {
+            if (_reloadInputAction != null)
+            {
+                _reloadInputAction.Disable();
+            }
+
+            if (_fireInputAction != null)
+            {
+                _fireInputAction.Disable();
+            }
+            
+            if (_swapWeaponInputAction != null)
+            {
+                _swapWeaponInputAction.Disable();
+            }
+
+            _isMouseScrollEnabled = false;
+        }
+
+        private void SetupWeapons()
+        {
+            foreach (var weapon in _weapons)
+            {
+                weapon.Initialize(_fireInputAction, _reloadInputAction);
+                weapon.enabled = false;
+            }
+
+            _weapons[_currentWeaponIndex].enabled = true;
+        }
+        
+        private void SetupPlayerEquipmentInput()
+        {
+            _reloadInputAction = _inputActions.FindAction("Reload");
+            _fireInputAction = _inputActions.FindAction("Fire");
+            _swapWeaponInputAction = _inputActions.FindAction("WeaponSwap");
+        }
+
+        private void SetupControllerSwapWeaponInputBinding()
+        {
+            _swapWeaponInputAction.started += OnControllerSwapWeaponAction;
+        }
+        
+        private void ScrollSwapWeapons(bool isUpwards)
+        {
+            _weapons[_currentWeaponIndex].enabled = false;
+            UpdateCurrentWeaponIndex(isUpwards);
+            _weapons[_currentWeaponIndex].enabled = true;
+        }
+        
+        private void UpdateCurrentWeaponIndex(bool isUpwards)
+        {
+            var indexLength = _weapons.Length - 1;
+            _currentWeaponIndex += isUpwards ? 1 : -1;
+            
+            if (_currentWeaponIndex > indexLength)
+            {
+                _currentWeaponIndex = 0;
+            } 
+            else if (_currentWeaponIndex < 0)
+            {
+                _currentWeaponIndex = indexLength;
+            }
+        }
+
+
+
+        private void Update()
+        {
+            OnMouseSwapWeaponAction();
+        }
+
+        private void Start()
+        {
+            Initialize();
+        }
+
+        private void Awake()
+        {
+            if (_inputActions == null)
+            {
+                Debug.LogWarning("Input Action Asset is missing from equipment behaviour");
+            }
+
+            if (_weapons.IsNullOrEmpty())
+            {
+                Debug.LogWarning("There are currently no weapons equipped within Equipment Behaviour");
+            }
+            
+            SetupPlayerEquipmentInput();
+        }
+        
+        #region Input 
+        private void OnMouseSwapWeaponAction()
+        {
+            var scrollVector = _isMouseScrollEnabled ? Mouse.current.scroll.ReadValue().normalized : Vector2.zero;
```

Comment:
@unoctium1 this is the only big problem i have with this task/new input system.
Mouse.current.scroll.ReadValue(), i can't for the life of my get the scroll wheel working.

Reply:
I'm not sure I understand, what you have seems to be working for me? It might be an issue for adding controller bindings but this works

Reply:
So instead of adding the "mouse scroll" within the New Input System, I'm called the Mouse directly. So I am not able to set it up through the Input Action Asset

Reply:
Did you see this thread? https://forum.unity.com/threads/how-do-you-get-mouse-scroll-input.825672/

Reply:
yup, looked that up and tried their settings.

Reply:
We could just call this good enough for now and then make a big ticket? It sounds like this might be an issue on Unity's end